File: backend/apps/observability/services/observability_service.py
# yss_orbit\backend\apps\observability\services\observability_service.py
from apps.observability.services.metrics_service import MetricsService
from apps.observability.services.trace_service import TraceService
from apps.observability.services.slo_service import SloService

class ObservabilityService:
    """
    Facade combining metrics, traces, and SLOs.
    """
    pass

File: frontend/src/features/observability/services/observabilityService.ts
// yss_orbit\frontend\src\modules\observability\services\observabilityService.ts
// @ts-expect-error - Auto-patched TS2307
import { BaseService } from '@/utils/core/api/baseService';
import { ObservabilityMetric } from '../types/observabilityTypes';

class ObservabilityApiService extends BaseService {
  constructor() {
    super('/observability');
  }

  getMetrics(): Promise<ObservabilityMetric[]> {
    // @ts-expect-error - Auto-patched TS2339
    return this.get<ObservabilityMetric[]>('/metrics/summary');
  }

  getTrace(traceId: string): Promise<any> {
    // @ts-expect-error - Auto-patched TS2339
    return this.get<any>(`/traces/${traceId}`);
  }
}

export const ObservabilityService = new ObservabilityApiService();
